File: PaymentSDK/UIHostObjCApp/views/SettingsViewController.m
//
//  SettingsViewController.m
//  UIHostObjCApp
//
//  Created by Rahul Dhuri on 06/06/19.
//  Copyright © 2019 Network International. All rights reserved.
//

#import "SettingsViewController.h"

@interface SettingsViewController ()

@end

@implementation SettingsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _currencies = [NSMutableArray arrayWithObjects: @"USD", @"AED", nil];
    
    _currencyPicker.delegate = self;
    _currencyPicker.dataSource = self;
    _emailInput.delegate = self;
    
    _emailInput.text = _email;
    [_emailInput addTarget:self action:@selector(textFieldDidChange:) forControlEvents: UIControlEventEditingChanged];
    NSUInteger row = [_currencies indexOfObject: _currency];
    [_currencyPicker selectRow: row inComponent: 0 animated: true];
}

-(void)textFieldDidChange:(UITextField *)textField {
    if ([textField.text length]) {
        [_delegate didEmailChange: textField.text];
    }
}

//MARK: UIPickerview datasource
-(NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
    return 1;
}

-(NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
    return [_currencies count];
}

//MARK: UIPickerview delegate
-(NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component {
    return _currencies[row];
}

-(void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component {
    [_delegate didCurrencyChange: _currencies[row]];
}

@end

File: PaymentSDK/UIHostApp/SettingsViewController.swift
//
//  SettingsViewController.swift
//  merchant-sample-app
//
//  Created by Niraj Chauhan on 4/24/19.
//  Copyright © 2019 Niraj Chauhan. All rights reserved.
//

import Foundation
import UIKit


class SettingsViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource, UITextFieldDelegate {
    
    weak var settingsDelegate: SettingsDelegate?
    
    @IBOutlet weak var emailInput: UITextField!
    @IBOutlet weak var currencyPicker: UIPickerView!
    
    let currencies = ["USD", "AED"]
    var email: String?
    var currency: String?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.currencyPicker.delegate = self
        self.currencyPicker.dataSource = self
        self.emailInput.delegate = self
        
        self.emailInput.text = email
        if let row = currencies.firstIndex(where: {$0 == currency}) {
            self.currencyPicker.selectRow(row, inComponent: 0, animated: true)
        }
        
        emailInput.addTarget(self, action: #selector(textFieldDidChange(_:)), for: .editingChanged)
    }
    
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 1
    }
    
    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return currencies.count
    }
    
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        return currencies[row]
    }
    
    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        settingsDelegate?.didCurrencyChange(currencies[row])
    }
    
    @objc func textFieldDidChange(_ textField: UITextField) {
        if let value = textField.text {
            settingsDelegate?.didEmailChange(value)
        }
    }

    
}
